napari/utils/events/event.py

```diff
@@ -258,6 +258,7 @@ def __init__(
         self._block_counter: Counter[Optional[Callback]] = Counter()
 
         # used to detect emitter loops
+        self._emitting = False
```

Comment:
No problem with these additions, but curious why they are needed - was there a bug before? Maybe you can add a comment in the code

Reply:
I needed them to avoid an emitter loop in [`selectionChanged`](https://github.com/napari/napari/pull/2556/files#diff-1d29b7b2ae740db911dd210ab47b659628e811dd428b5aa51ea3517fb23cbe1bR77).  this variable is actually present in the [original vispy events.py](https://github.com/vispy/vispy/blob/184d854c7542f4c6df21bb4c7db3dd5b340dd900/vispy/util/event.py#L431) ... it must have gotten removed from our version, but I think it's actually useful

Reply:
hmm funny, not sure what happened there!